core/src/test/java/com/linecorp/armeria/client/HttpClientMaxConcurrentStreamTest.java

```diff
@@ -168,4 +177,225 @@ public void connectionClosed(SessionProtocol protocol, InetSocketAddress remoteA
             assertThat(closes).hasValue(0);
         }
     }
+
+    @Test
+    void handleExceedsMaxStreamsBasicCase() throws Exception {
+        final List<ClientConnectionTimings> connectionTimings = new ArrayList<>();
+        final WebClient client = WebClient.builder(server.uri(SessionProtocol.H2C, "/"))
+                                          .factory(clientFactory)
+                                          .decorator((delegate, ctx, req) -> {
+                                              ctx.logBuilder().whenAvailable(RequestLogProperty.SESSION)
+                                                 .thenAccept(requestLog -> {
+                                                     connectionTimings.add(requestLog.connectionTimings());
+                                                 });
+                                              return delegate.execute(ctx, req);
+                                          })
+                                          .build();
+        final List<CompletableFuture<AggregatedHttpResponse>> receivedResponses = new ArrayList<>();
+        final int numRequests = MAX_CONCURRENT_STREAMS + 1;
+
+        clientFactory.eventLoopGroup().execute(() -> {
+            for (int j = 0; j < numRequests; j++) {
```

Comment:
- `j` could be `i`?
- Could you add a comment about why you need to execute this via `.execute()`?